karapace/backup/backends/v3/backend.py

```diff
@@ -168,10 +203,16 @@ def get_checkpoint(
             return self.running_checksum.digest()
         return None
 
-    def increase(self, bytes_offset: int) -> None:
+    def update(
+        self,
+        bytes_offset: int,
+        record_offset: int,
+    ) -> None:
         self.records_written_since_checkpoint += 1
         self.records_written += 1
         self.bytes_written_since_checkpoint += bytes_offset
+        self.min_offset = record_offset if self.min_offset is None else min(record_offset, self.min_offset)
+        self.max_offset = record_offset if self.max_offset is None else max(record_offset, self.max_offset)
```

Comment:
Is it not guaranteed that `record_offset` > `self.min_offset` ?

Reply:
You're right, if these aren't monotonic it means that something is broken. It probably makes sense to be rigid about that, and raise errors should these expectations not hold.